src/export/ValueSetExporter.ts

```diff
@@ -86,6 +86,14 @@ export class ValueSetExporter {
           composeElement.valueSet = component.from.valueSets.map(vs => {
             return this.fisher.fishForMetadata(vs, Type.ValueSet)?.url ?? vs;
           });
+          composeElement.valueSet = composeElement.valueSet.filter(vs => {
+            if (vs == valueSet.url) {
+              logger.error(
+                `Value set with id ${valueSet.id} has component rule with self referencing value set (by id, value set name, or url). Skipping rule.`
+              );
```

Comment:
I'm not sure if this error message is completely accurate. If you have a rule on a ValueSet like this:

```
ValueSet: MaxVS
Id: max-vs
* include codes from valueset MaxVS and OtherVS
```
the filter will filter out the `MaxVS` reference, but _keep_ the OtherVS reference. So the rule itself isn't skipped, we're only skipping _part_ of the rule.

That said, I actually am not sure what we want to do in this scenario. If someone did use this rule, I'm not sure if we want to just filter out `MaxVS` because if they asked for the intersection of codes between MaxVS and OtherVS and we end up giving them just all the codes in OtherVS, is that the behavior we want? If it is, we should update the error message to just say "skipping this value set" rather than "skipping rule". If it isn't and we decide that we're making it so SUSHI is giving back something unexpected and not what an author is trying to do, we probably want to fully skip the rule if _any_ of the value sets are self referencing.

Reply:
After talking through this in person, we decided that the behavior is correct (skip over MaxVS and use any remaining value sets) but we should update the error message.

Perhaps something along the lines of "removing self-reference" (or some better wording you think of).